server/src/services/user.service.ts

```diff
@@ -0,0 +1,55 @@
+import { UpdateUserBody } from '../types/user.types';
+import { UserRepository } from '../repositories/user.repository';
+import { History, PopulatedHistory } from '../models/history.model';
+import { SavedPackage, PopulatedSavedPackage } from '../models/saved-packages.model';
+import { HistoryRepository } from '../repositories/history.repository';
+import { SavedPackageRepository } from '../repositories/saved-packages.repository';
+import mongoose from 'mongoose';
+import { populateAggregation } from '../queries/package.query';
+
+export const UserService = {
+    async updateUserById(
+        userId: string,
+        data: UpdateUserBody
+    ): Promise<Omit<any, 'password' | 'refreshTokens'> | null> {
```

Comment:
```suggestion
Promise<Omit<User, 'password' | 'refreshTokens'> | null> 
```
I believe you wanted to make it User without `password` and `refreshTokens`. you can use `PublicUser type` that for sure exist in client and maybe even in the server.